core-primitives/test/src/mock/trusted_operation_pool_mock.rs

```diff
@@ -154,21 +151,22 @@ impl TrustedOperationPool for TrustedOperationPoolMock {
 		unimplemented!()
 	}
 
+	#[allow(clippy::type_complexity)]
 	fn ready(
 		&self,
 		shard: ShardIdentifier,
 	) -> Box<dyn Iterator<Item = Arc<Self::InPoolOperation>> + Send> {
 		let transactions = self.submitted_transactions.read().unwrap();
 		let ready_transactions = transactions
 			.get(&shard)
-			.map(|payload| payload.xts.iter().map(|xt| Self::map_stf_top_to_tx(xt)).collect())
-			.unwrap_or_else(|| vec![]);
+			.map(|payload| payload.xts.iter().map(Self::map_stf_top_to_tx).collect())
+			.unwrap_or_else(Vec::new);
 		Box::new(ready_transactions.into_iter())
 	}
 
 	fn shards(&self) -> Vec<ShardIdentifier> {
 		let transactions = self.submitted_transactions.read().unwrap();
-		transactions.iter().map(|(shard, _)| shard.clone()).collect()
+		transactions.iter().map(|(shard, _)| *shard).collect()
```

Comment:
Again, with `into_iter` you can prevent implicit copies.

Reply:
Again, we do have to make copies, we're not returning references 😇. Or am I missing something?

Reply:
(and `HashMap` does not implement `into_iter()` (which would make copies anyway?) )

Reply:
Ah ok, maybe I got it wrong, and you can't use it here. I thought that `transactions` is an owned value, which is why `into_iter` does not `copy` but consumes the object. Hence, you can get owned values without copy or clone.

But it seems that is not possible here with the hashmap. Thanks for checking it out. Learned something new. :)

Reply:
No, unfortunately the cases where you suggested this, are all `self` values, so we cannot move them, but have to return copies (which is not a problem, since these structures are cheap to copy and part of mocks anyway ☺️ )

Reply:
Yes, sorry for the inconvenience in that case. :)

Reply:
No worries, these are good inputs - I definitely learnt something 👍